node/heartbeat/monitor.go

```diff
@@ -236,6 +247,14 @@ func (m *Monitor) ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedPe
 		return err
 	}
 
+	isHardforkTrigger := bytes.Equal(hbRecv.Payload, []byte(hardforkTriggerString))
+	if isHardforkTrigger {
+		errHardforkTrigger := m.hardforkTrigger.TriggerReceived(hbRecv.Pubkey, hbRecv.Payload)
```

Comment:
Question, why is the `hardforkTrigger` check inside the heartbeat? It seems to me that it receives more functionality than it was intended.

Reply:
Agree, couldn't find another (easier) way to do this. I have re-purposed the heartbeat topic to send this special trigger message whenever it is necessary. Why? Because it has all the ingredients I needed: global topic + interceptors + antiflood capabilities + message signing verification. It was the best place I could find to make the injection here.